File: examples/quicksort.py
from sys import stdin

@@ begin hide
def main():
    print("start qs", stdin.readline())  # Ignore first line with number of inputs on it
    array_in = stdin.readline()
    print(array_in)
    presort = list(map(int, array_in.split(' ')))
    sort = quickSort(presort)
    print(sort)
@@ end hide

def quickSort(arr):
    less = []
    pivotList = []
    more = []
    if len(arr) <= 1:
@@ begin question base case
@@ description: Find the base case
@@ points: 100
        return arr
@@ end question
    else:
        pivot = arr[0]
@@ begin question recursion
@@ points: 500
        for i in arr:
            if i < pivot:
                less.append(i)
            elif i > pivot:
                more.append(i)
            else:
                pivotList.append(i)
@@ end question
        less = quickSort(less)
        more = quickSort(more)
        return less + pivotList + more

@@ begin hide
if __name__ == '__main__':
    main()
@@ end hide

File: Examples/quicksort.hs
@@ begin hide
main = do
	getLine
	xs <- getLine >>= return . map read . unwords
	print $ qsort xs
@@ begin hide

@@ begin problem
@@ description: Why does anyone like haskell?
@@ begin question quicksort
@@ points: 100000
qsort :: [Int] -> [Int]
qsort [] = []
qsort (x:xs) = qsort [y | y <- xs, y < x] ++ [x] ++ qsort [y | y <- xs, y >= x]
@@ end question
@@ end problem